clc/modules/msgs/src/main/java/com/eucalyptus/ws/Handlers.java

```diff
@@ -175,7 +176,47 @@ private static NioServerHandler newNioServerHandler( ) {
   }
 
   private static ChannelHandler newChunkedWriteHandler( ) {
-    return new ChunkedWriteHandler( );
+    return new ChunkedWriteHandler( ) {
+      private void flush( final ChannelHandlerContext ctx ) {
+        try {
+          beforeRemove( ctx );
+        } catch ( final Exception e ) {
+          LOG.warn( "Unexpected exception while sending chunks.", e );
+        }
+      }
+
+      @Override
+      public void handleUpstream( final ChannelHandlerContext ctx, final ChannelEvent e ) throws Exception {
+        if ( e instanceof ChannelStateEvent ) {
+          final ChannelStateEvent cse = (ChannelStateEvent) e;
+          if ( INTEREST_OPS == cse.getState( ) ) {
+            // flush off the io thread to avoid blocking other channels
+            pipelineExecutionHandler( ).getExecutor( ).execute( () -> flush( ctx ) );
+            ctx.sendUpstream(e);
+            return;
+          }
+        }
+        super.handleUpstream( ctx, e );
+      }
```

Comment:
This is the super impl:

```
    public void handleUpstream(ChannelHandlerContext ctx, ChannelEvent e)
            throws Exception {
        if (e instanceof ChannelStateEvent) {
            ChannelStateEvent cse = (ChannelStateEvent) e;
            switch (cse.getState()) {
            case INTEREST_OPS:
                // Continue writing when the channel becomes writable.
                flush(ctx, true);
                break;
            case OPEN:
                if (!Boolean.TRUE.equals(cse.getValue())) {
                    // Fail all pending writes
                    flush(ctx, true);
                }
                break;
            }
        }
        ctx.sendUpstream(e);
    }
```

so for INTEREST_OPS we now call "flush" on a different thread, anything else still goes to super.